### PR TITLE
Fix blob creation in trino file system tests

### PR DESCRIPTION
**Pull request by elonazoulay**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by elonazoulay** [comment]
Note: if the root location is not empty then `testDirectoryExists` was double appending the path when calling `createBlob` - lmk if this fix is ok.

**Comment by elonazoulay** [comment]
I can also add the `RemoteStorageHelper` back, I will need the project id used for ci tests - either way `createBlob` needs to be changed (comment above).

File: lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java [hunk 1]
```diff
@@ -880,7 +881,7 @@ public void testDirectoryExists()
             }
             else {
                 assertThat(getFileSystem().directoryExists(createLocation(directoryName))).isEmpty();
-                createBlob(closer, createLocation(directoryName).appendPath(fileName).path());
+                createBlob(closer, "%s/%s".formatted(directoryName, fileName));
```
Comment (elonazoulay) [inline lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java:884]: Should line 877 be changed as well? i.e. for the equivalent `createBlob` for hierarchical filesystems.

File: lib/trino-filesystem-gcs/src/test/java/io/trino/filesystem/gcs/AbstractTestGcsFileSystem.java [hunk 2]
```diff
@@ -57,22 +55,21 @@ protected void initialize(String gcpCredentialKey)
...
         byte[] jsonKeyBytes = Base64.getDecoder().decode(gcpCredentialKey);
         GcsFileSystemConfig config = new GcsFileSystemConfig().setJsonKey(new String(jsonKeyBytes, UTF_8));
-        GcsStorageFactory storageFactory = new TestingGcsStorageFactory(config);
+        GcsStorageFactory storageFactory = new DefaultGcsStorageFactory(config);
         this.gcsFileSystemFactory = new GcsFileSystemFactory(config, storageFactory);
         this.storage = storageFactory.create(ConnectorIdentity.ofUser("test"));
-        String bucket = RemoteStorageHelper.generateBucketName();
-        storage.create(BucketInfo.of(bucket));
-        this.rootLocation = Location.of("gs://%s/".formatted(bucket));
+        if (storage.get(bucket) == null) {
```
Comment (elonazoulay) [inline lib/trino-filesystem-gcs/src/test/java/io/trino/filesystem/gcs/AbstractTestGcsFileSystem.java:61]: Should I just check for the existence of the bucket and not try to create it?

**Comment by electrum** [comment]
/test-with-secrets sha=08595ccc84d539eadba1fe012114f25181b79e6c

**Review by electrum — CHANGES_REQUESTED** [review]
We should go with the original approach of creating the bucket in the test, since we want to test that the code works with the root of the bucket. This is what we do for S3 and Azure.

**Pull request closed without merging**